packages/medusa/src/services/custom-shipping-option.ts

```diff
@@ -0,0 +1,104 @@
+import { MedusaError } from "medusa-core-utils"
+import { EntityManager } from "typeorm"
+import { TransactionBaseService } from "../interfaces"
+import { CustomShippingOption } from "../models"
+import { CustomShippingOptionRepository } from "../repositories/custom-shipping-option"
+import { FindConfig, Selector } from "../types/common"
+import { CreateCustomShippingOptionInput } from "../types/shipping-options"
+import { buildQuery } from "../utils"
+
+class CustomShippingOptionService extends TransactionBaseService<CustomShippingOptionService> {
+  protected manager_: EntityManager
+  protected transactionManager_: EntityManager | undefined
+  protected customShippingOptionRepository_: typeof CustomShippingOptionRepository
+
+  constructor({ manager, customShippingOptionRepository }) {
```

Comment:
suggestion: If you could add some typings such as `InjectedDependencies` to keep the consistency